stranding/stranding.py

```diff
@@ -0,0 +1,183 @@
+        elif chr_name in ('0', 0):
+            raise Unstrandable('Chromosome 0 is unmapped')
+        elif max(len(_5p), len(_3p)) < self.min_flank_length:
+            raise FlanksTooShort('At least one flank must be longer than the specified'
+                                 'minimum flank length of s %d' % self.min_flank_length)
+
+        # chromosome-specific conventions
+        loop = chr_name == 'MT'
+        chr_name = chr_name if chr_name != 'XY' else 'X'
+        max_length = max(len(_5p), len(_3p))
+
+        # reference sequences
+        try:
+            chromosome = Chromosome(chr_name, build, loop=loop)
+            ref_5p = chromosome.sequence(pos - window - max_length, pos + window)
+            ref_3p = chromosome.sequence(pos - window, pos + max_length + window + 1)
+        except ValueError:
+            raise MissingReferenceFlank(
+                'Could not find flanks for %s %d %d' % (chr_name, pos, window))
+
+	# exact comparisons are cheap so try this first
+        if window == 0 and (_3p == ref_3p or _5p == ref_5p):
+            return 1
+
+        ref_5p_RC = str(Seq(ref_5p).reverse_complement())
+        ref_3p_RC = str(Seq(ref_3p).reverse_complement())
+
+        if window == 0 and (_3p == ref_5p_RC or _5p == ref_3p_RC):
+            return -1
+
+        if window == 0 and self.tolerance == 1.0:
+            raise Unstrandable('Strict stranding failed')
+
+        # alignments are expensive so try to do as few as possible
+        fwd_5p_score = self.align(ref_5p, _5p)
+        if self.is_perfect_score(fwd_5p_score, _5p):
+            return 1
+
+        fwd_3p_score = self.align(ref_3p, _3p)
+        if self.is_perfect_score(fwd_3p_score, _3p):
+            return 1
+
+        rev_5p_score = self.align(ref_5p_RC, _3p)
+        if self.is_perfect_score(rev_5p_score, _3p):
+            return -1
+
+        rev_3p_score = self.align(ref_3p_RC, _5p)
+        if self.is_perfect_score(rev_3p_score, _5p):
+            return -1
+
+        strands = []
+        if self.is_high_scoring(fwd_5p_score, _5p):
+            strands.append(1)
+        if self.is_high_scoring(fwd_3p_score, _3p):
+            strands.append(1)
+        if self.is_high_scoring(rev_5p_score, _3p):
+            strands.append(-1)
+        if self.is_high_scoring(rev_3p_score, _5p):
+            strands.append(-1)
+
+        if len(set(strands)) > 1:
+            # Alignments were accepted on both strands (!)
+            # The flanks may be too short or the tolerance may be too loose.
+            LOGGER.error('Forward alignments')
+            self.align_and_log(ref_5p, _5p)
+            self.align_and_log(ref_3p, _3p)
+            LOGGER.error('Reverse alignments')
+            self.align_and_log(ref_5p_RC, _3p)
+            self.align_and_log(ref_3p_RC, _5p)
+            raise InconsistentAlignment('Inconsistent alignments')
+        elif strands:
+            return strands[0]
+        raise Unstrandable('No matching alignments')
```

Comment:
Something like this might be clearer:

```
is_fwd = self.is_high_scoring(fwd_5p_score, _5p) or self.is_high_scoring(fwd_3p_score, _3p)
is_rev = self.is_high_scoring(rev_5p_score, _3p) or self.is_high_scoring(rev_3p_score, _5p)

if is_fwd and is_rev:
    # handle error
elif is_fwd:
    return 1
elif is_rev:
    return -1
else:
    raise Unstrandable
```

Also it might be nice to define `FORWARD_STRAND = 1` etc.